### PR TITLE
[compiler-rt][rtsan] NFC: Refactor context helper functions

### PR DESCRIPTION
**Pull request by cjappl**
Pull `ExpectNotRealtime` and `PrintDiagnostics` into their own functions, away from Context.

Why?
- Currently PrintDiagnostics does not depend on any private method of Context
- We want to change soon to allow for detection of infinite loops and user defined blocking functions
    - These changes will affect these two functions, but shouldn't affect Context.
    - We will need to add an argument, or otherwise change these methods to say "why" we expect not realtime in this situation.

Overall, I think this refactor decreases coupling, and increases our flexibility going forward.

### CODE REVIEW

**Comment by cjappl** [comment]
CC @davidtrevelyan for review

File: compiler-rt/lib/rtsan/rtsan_context.h [hunk 1]
```diff
@@ -22,17 +22,22 @@ class Context {
...
   bool IsBypassed() const;
-  void PrintDiagnostics(const char *intercepted_function_name);
 
+  Context(const Context &) = delete;
+  Context(Context &&) = delete;
+  Context &operator=(const Context &) = delete;
+  Context &operator=(Context &&) = delete;
+
+private:
   int realtime_depth{0};
```
Comment (davidtrevelyan) [inline compiler-rt/lib/rtsan/rtsan_context.h:34]: Minor style nit: private members I think should have trailing underscores :)

**Comment by davidtrevelyan** [comment]
LGTM!

**Comment by cjappl** [comment]
Test failure is unrelated:

```
_bk;t=1725135284392FAIL: lld :: ELF/avr-reloc.s (83333 of 85610)

_bk;t=1725135284392******************** TEST 'lld :: ELF/avr-reloc.s' FAILED ********************

_bk;t=1725135284392Exit Code: 1

_bk;t=1725135284392

_bk;t=1725135284392Command Output (stderr):
```

**Pull request merged**